In [1]:
import pandas as pd

df = pd.read_csv('/Users/vedharai/hr-attrition-predictor/data/hr_clean.csv')
print(df.shape)

(1470, 45)


In [2]:
X = df.drop(columns=['Attrition'])
y = df['Attrition']
print(X.shape)
print(y.shape)

(1470, 44)
(1470,)


In [5]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [15]:
from sklearn.ensemble import RandomForestClassifier

rf_model = RandomForestClassifier(n_estimators=100, class_weight='balanced', random_state=42, max_depth=3, min_samples_leaf=10)
rf_model.fit(X_train, y_train)
predictions = rf_model.predict(X_test)

In [16]:
from sklearn.metrics import classification_report
print(classification_report(y_test, predictions))

              precision    recall  f1-score   support

           0       0.92      0.84      0.88       255
           1       0.33      0.51      0.40        39

    accuracy                           0.80       294
   macro avg       0.63      0.68      0.64       294
weighted avg       0.84      0.80      0.82       294



Random forest findings: Max attrition class recall obtained by manually tuning the model is 0.51. Logistic Regression is still beating the complex model will 0.59 class 1 recall. 